zerver/lib/presence.py

```diff
@@ -48,36 +48,18 @@ def get_modern_user_presence_info(
 def get_legacy_user_presence_info(
     last_active_time: datetime.datetime, last_connected_time: datetime.datetime
 ) -> Dict[str, Any]:
-    # Reformats the modern UserPresence data structure so that legacy
-    # API clients can still access presence data.
-    #
-    # We expect this code to remain mostly unchanged until we can delete it.
-
-    if timezone_now() - last_active_time > datetime.timedelta(minutes=2):
-        dt = last_connected_time
-        status = UserPresence.LEGACY_STATUS_IDLE
-    else:
-        dt = last_active_time
-        status = UserPresence.LEGACY_STATUS_ACTIVE
-
-    client_name = "website"
-    timestamp = datetime_to_timestamp(dt)
-
-    # This field was never used by clients of the legacy API, so we
-    # just set it to a fixed value for API format compatibility.
-    pushable = False
```

Comment:
Did this deduplication to resolve https://github.com/zulip/zulip/pull/16381#discussion_r716084830